ec/ecresource/deploymentresource/deployment/v2/use_null_unless_adding_apm_plan_modifier.go

```diff
@@ -0,0 +1,99 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
```

Comment:
nitpick: plan modifier files are named inconsistently, e.g. this file name doesn't correspond to the plan modifier name that stands true for `setUnknownIfResetPasswordIsTrue plan modifier.